sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -120,14 +120,14 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
         return -EINVAL;
     }
 #if IS_USED(MODULE_ZTIMER_USEC)
-    ztimer_t timeout_timer;
+    ztimer_t timeout_timer = { .base = { .next = NULL } };
```

Comment:
```suggestion
    ztimer_t timeout_timer = {.callback = _callback_put};
```

Reply:
The callback is only needed for when the if is set below. So I would rather not initialize something that is not needed and go for the `NULL` initializer as picked.

Reply:
yeah, maybe be a good example.

https://github.com/kaspar030/RIOT/blob/ef4b453d4dde0122dee7fcc511f474d5a09f0750/sys/include/ztimer.h#L76-L82